stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,422 @@
+// Swift directly. Instead, users should import 'Android' or another
+// umbrella module that includes these headers.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The top level 'Bionic' module is the C standard library
+// used in the Android NDK.
+module Bionic [system] {
+  header "SwiftBionic.h"
+  export *
+}
+
+// The individual modules for the Bionic C standard library used
+// by the Android NDK.
+module _bionic_assert [system] {
+  // <assert.h>'s use of NDEBUG requires textual inclusion.
+  textual header "assert.h"
+}
+module _bionic_complex [system] {
+  header "complex.h"
+  export *
+}
+module _bionic_ctype [system] {
+  header "ctype.h"
+  export *
+}
+module _bionic_errno [system] {
+  header "errno.h"
+  export *
+}
+module _bionic_fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers.
+// FIXME: this can be dropped once NDK's libc++ 'std' module is split
+// into individual submodules.
+module _bionic_inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use _bionic_stdint
+  use _android_sys_cdefs
+}
+module _bionic_limits [system] {
+  header "limits.h"
+  header "bits/posix_limits.h"
+  export *
+}
+module _bionic_locale [system] {
+  header "locale.h"
+  export *
+}
+module _bionic_malloc [system] {
+  header "malloc.h"
+  export *
+}
+module _bionic_math [system] {
+  header "math.h"
+  export *
+  link "m"
+}
+module _bionic_pty [system] {
+  header "pty.h"
+  export *
+}
+module _bionic_setjmp [system] {
+  header "setjmp.h"
+  export *
+}
+module _bionic_signal [system] {
+  header "signal.h"
+  // The 'signal.h' header unfortunately has a circular include
+  // with 'sys/ucontext.h' and thus it must be part of this module.
+  header "sys/ucontext.h"
+  export *
+}
+module _bionic_stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+module _bionic_stdint [system] {
+  header "stdint.h"
+  export *
+}
+module _bionic_stdio [system] {
+  header "stdio.h"
+  export *
+}
+module _bionic_stdio_ext [system] {
+  header "stdio_ext.h"
+  export *
+}
+module _bionic_stdlib [system] {
+  header "stdlib.h"
+  export *
+}
+module _bionic_string [system] {
+  header "string.h"
+  export *
+}
+module _bionic_time [system] {
+  header "time.h"
+  export *
+}
+module _bionic_utmp [system] {
+  header "utmp.h"
+  export *
+}
+
+// POSIX and another android NDK headers.
+module _android_alloca [system] {
+  header "alloca.h"
+  export *
+}
+
+module _android_core [system] {
+  header "ar.h"
+  header "cpio.h"
+  header "dirent.h"
+  header "dlfcn.h"
+  header "err.h"
+  header "errno.h"
+  header "fcntl.h"
+  header "fnmatch.h"
+  header "fts.h"
+  header "glob.h"
+  header "mntent.h"
+  header "libgen.h"
+  header "nl_types.h"
+  header "paths.h"
+  header "poll.h"
+  header "pwd.h"
+  header "spawn.h"
```

Comment:
Can we do better with the grouping? like spawn.h seems unrelated to tar.h, but mbstate is by standard part of wchar. arpa/inet.h is decently related to linux/if.h too, etc etc.